stubs/Flask-SQLAlchemy/flask_sqlalchemy/__init__.pyi

```diff
@@ -40,7 +39,9 @@ class Pagination:
         self, left_edge: int = ..., left_current: int = ..., right_current: int = ..., right_edge: int = ...
     ) -> None: ...
 
-class BaseQuery(_Query):
+_T = TypeVar("_T")
+
+class BaseQuery(Query[_T]):
```

Comment:
We should also make this generic, like the base class:

```
class BaseQuery(Query[_T], Generic[_T]):
```

(Needs `Generic` import from `typing`.)

Reply:
That's not necessary, inheriting from a generic base that is not fully specialized is enough to make the class generic.